packages/web/hooks/language/context.tsx

```diff
@@ -0,0 +1,73 @@
+import { PropsWithChildren, useCallback, useState } from "react";
+
+import en from "~/localizations/en.json";
+import { createContext } from "~/utils/react-context";
+
+import { internalTranslate } from "./internal";
+import {
+  MultiLanguageContextState,
+  MultiLanguageT,
+  Translations,
+} from "./types";
+
+const [MultiLanguageInnerProvider, useMultiLanguage] =
+  createContext<MultiLanguageContextState>({
+    strict: true,
+    name: "LanguageContext",
+  });
+
+interface MultiLanguageProviderProps {
+  defaultLanguage?: string;
+  defaultTranslations?: Translations;
+}
+
+const DEFAULT_LANGUAGE = "en";
+
+let globalTranslations: Translations = { en };
+let globalLanguage: string = DEFAULT_LANGUAGE;
+
+export const t: MultiLanguageT = (path, args) =>
```

Comment:
why is the `t` function declared twice?

Reply:
Because there are some parts of the application, which access "t" without being React components, for example, the function that creates a "toast" does not have direct access to the react context